types/haversine/index.d.ts

```diff
@@ -1,13 +1,24 @@
-// Type definitions for haversine 1.0
+// Type definitions for haversine 1.1
 // Project: https://github.com/njj/haversine
 // Definitions by: Christian Rackerseder <https://github.com/screendriver>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare namespace haversine {
-    interface Coordinate {
+    type LatLonTuple = [number, number];
+
+    interface GeoJSON {
+        geometry: {
+            coordinates: LatLonTuple
+        };
+    }
+
+    type Coordinate = ({
```

Comment:
When you remove the interface `Coordinate` and make instead a union type with object literals it will be impossible to determine the type if you have a function or variable that gets passed through your application. For example:

```js
function foo(coord: haversine.Coordinate) {
  return coord.longitude; // Which type has haversine.Coordinate?
}
```

produces the compile error `Property 'longitude' does not exist on type '{ lat: number; lon: number; }'.` Because it is a union type. The solution here would be a type guard

```js
function isLongitude(coord: haversine.Coordinate): coord is ??? {
  if ((coord as ??? ).longitude) {
    // ...
  }
  // ...
}
```

But as you can see I marked every place that does not work with `???`. I could replace `coord is ???` with my own object literal but then I have
1. code duplication (because the right types are already defined here in this type definition)
1. not a named type from this type definition.

So now it is not possible to pass a `Coordinate` variable through my functions and do anything with it. The solution would be to extract every anonymous object literal type to it's own interface declaration

Reply:
Ah, thank you very much for the explanation - makes sense. I'll make the changes